### ISSUE TITLE
Serve: Warn when using an `https://...` proxy source without a valid certificate

### ISSUE THREAD

**Issue by shayne** [tailscale/tailscale#8421]
Also, teach folks about `https+insecure` as an option.

**Comment by ruifung** [comment]
Yeah, just hit this and it would be really nice if this was in the docs.

Kind of hard to have a valid cert for 127.0.0.1 after all.

**Comment by nshalman** [comment]
Just to be very explicit for anyone doing what I should have done and searching the open issues for this...
If you're doing something like trying to forward TLS traffic to local TLS server using a self-signed certificate, if you see log messages like:
```
http: proxy error: tls: failed to verify certificate: x509: cannot validate certificate for 127.0.0.1 because it doesn't contain any IP SANs
```
Then you probably want to use `https+insecure` e.g.
```
tailscale serve https:443 / https+insecure://localhost:8443
```

**Comment by dehnert** [comment]
> Then you probably want to use https+insecure e.g.
> tailscale serve https:443 / https+insecure://localhost:8443

It would be great if there were a way to say "do the DNS lookup for $dns, but send host headers, SNI, and verify the cert for $tls instead". `curl` supports this with the `--connect-to` option. In my case, I have a service running on localhost with a real cert, and I'd love to be able to connect to localhost but still verify the cert. Doing name-based vhosting but using `tailscale serve` to expose on separate VIPs is another use-case.

**Comment by nshalman** [comment]
> It would be great if there were a way to say "do the DNS lookup for $dns, but send host headers, SNI, and verify the cert for $tls instead". `curl` supports this with the `--connect-to` option. In my case, I have a service running on localhost with a real cert, and I'd love to be able to connect to localhost but still verify the cert. Doing name-based vhosting but using `tailscale serve` to expose on separate VIPs is another use-case.

+1 to that idea.

Also, reflecting on this years later, it would probably provide an improved user experience for the command setting up the serve or the funnel to probe whether the newly set up endpoint works as expected or not and provided an immediate warning if not rather than having to do external testing and go looking for the logs. I don't know how complex the code would be, though.
